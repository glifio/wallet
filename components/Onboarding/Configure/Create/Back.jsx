import React from 'react'
import PropTypes from 'prop-types'

import { Box, Button, Card, Text, Glyph } from '../../../Shared'
import useReset from '../../../../utils/useReset'

const Back = ({ setReturningHome }) => {
  const resetState = useReset()
  return (
    <Box display='flex' flexDirection='column' alignItems='center' mt={8}>
      <Card
        display='flex'
        flexDirection='column'
        justifyContent='space-between'
        width='100%'
      >
        <Glyph acronym='Bk' />
        <Text>Do you really wish to exit onboarding?</Text>
        <Text>You will be taken back to the home screen.</Text>
      </Card>
      <Box mt={6} display='flex' justifyContent='space-between' width='100%'>
        <Button
          title='No, keep onboarding'
          onClick={() => setReturningHome(false)}
          variant='secondary'
          mr={2}
        />
        <Button
          title='Yes, take me home'
          onClick={() => resetState()}
          variant='primary'
          ml={2}
        />
      </Box>
    </Box>
  )
}

Back.propTypes = {
  setReturningHome: PropTypes.func.isRequired
}

export default Back
