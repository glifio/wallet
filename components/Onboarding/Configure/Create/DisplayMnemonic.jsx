import React from 'react'
import { Text } from '../../../Shared'
import { MNEMONIC_PROPTYPE } from '../../../../customPropTypes'

const DisplayMnemonic = ({ mnemonic }) => {
  return (
    <>
      <Text color='core.primary'>{mnemonic}</Text>
    </>
  )
}

DisplayMnemonic.propTypes = {
  mnemonic: MNEMONIC_PROPTYPE
}

export default DisplayMnemonic
