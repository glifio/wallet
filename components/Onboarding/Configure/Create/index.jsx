import React, { useEffect, useState } from 'react'
import { number } from 'prop-types'
import { useDispatch } from 'react-redux'
import { useRouter } from 'next/router'
import Filecoin from '@openworklabs/filecoin-wallet-provider'
import {
  Box,
  Button,
  OnboardCard,
  StepHeader,
  LoadingScreen
} from '../../../Shared'

import Walkthrough from './Walkthrough'
import Back from './Back'
import { useWasm } from '../../../../lib/WasmLoader'
import { walletList } from '../../../../store/actions'
import { useWalletProvider } from '../../../../WalletProvider'
import { createWalletProvider } from '../../../../WalletProvider/state'
import reportError from '../../../../utils/reportError'

// we pass this optional prop to make testing the core wallet functionality easier
const Create = ({ initialWalkthroughStep }) => {
  const router = useRouter()
  const [mnemonic, setMnemonic] = useState('')
  const [walkthroughStep, setWalkthroughStep] = useState(initialWalkthroughStep)
  const [loading, setLoading] = useState(true)
  const [returningHome, setReturningHome] = useState(false)
  const [canContinue, setCanContinue] = useState(false)
  const [importSeedError, setImportSeedError] = useState(false)
  const { generateMnemonic } = useWasm()
  const {
    dispatch,
    fetchDefaultWallet,
    walletSubproviders: { HDWalletProvider }
  } = useWalletProvider()

  const dispatchRdx = useDispatch()

  const nextStep = () => {
    setImportSeedError(false)
    if (walkthroughStep === 1) setWalkthroughStep(2)
    else if (walkthroughStep === 2 && canContinue) setWalkthroughStep(3)
    else if (walkthroughStep === 2) setImportSeedError(true)
    else if (walkthroughStep >= 3) setWalkthroughStep(walkthroughStep + 1)
  }

  useEffect(() => {
    setMnemonic(generateMnemonic())
    setLoading(false)
  }, [setMnemonic, generateMnemonic])

  useEffect(() => {
    const instantiateProvider = async () => {
      try {
        const provider = new Filecoin(HDWalletProvider(mnemonic), {
          apiAddress: process.env.LOTUS_NODE_JSONRPC
        })
        dispatch(createWalletProvider(provider))
        const wallet = await fetchDefaultWallet(provider)
        dispatchRdx(walletList([wallet]))
        const params = new URLSearchParams(router.query)
        router.push(`/home?${params.toString()}`)
      } catch (err) {
        reportError(16, false, err.message, err.stack)
        setImportSeedError(err.message || JSON.stringify(err))
      }
    }
    if (walkthroughStep === 4) instantiateProvider()
  }, [
    dispatch,
    dispatchRdx,
    fetchDefaultWallet,
    mnemonic,
    router,
    walkthroughStep,
    HDWalletProvider
  ])

  return (
    <>
      {!returningHome ? (
        <>
          {loading || walkthroughStep === 4 ? (
            <LoadingScreen />
          ) : (
            <>
              <Box
                display='flex'
                flexDirection='column'
                justifyContent='center'
                alignItems='center'
              >
                <OnboardCard
                  display='flex'
                  flexDirection='row'
                  flexWrap='wrap'
                  justifyContent='center'
                  maxWidth={16}
                >
                  <StepHeader
                    currentStep={walkthroughStep}
                    totalSteps={3}
                    glyphAcronym='Sp'
                  />
                  {mnemonic && (
                    <Walkthrough
                      importSeedError={importSeedError}
                      canContinue={canContinue}
                      walkthroughStep={walkthroughStep}
                      mnemonic={mnemonic}
                      setCanContinue={setCanContinue}
                    />
                  )}
                </OnboardCard>
                <Box
                  mt={6}
                  display='flex'
                  width='100%'
                  justifyContent='space-between'
                >
                  <Button
                    title='Back'
                    onClick={() => {
                      if (walkthroughStep === 2) setWalkthroughStep(1)
                      else setReturningHome(true)
                    }}
                    variant='secondary'
                    mr={2}
                  />
                  <Button
                    title={
                      walkthroughStep === 1
                        ? "I've recorded my seed phrase"
                        : 'Next'
                    }
                    onClick={nextStep}
                    variant='primary'
                    ml={2}
                  />
                </Box>
              </Box>
            </>
          )}
        </>
      ) : (
        <Back setReturningHome={setReturningHome} />
      )}
    </>
  )
}

Create.propTypes = {
  initialWalkthroughStep: number
}

Create.defaultProps = {
  initialWalkthroughStep: 1
}

export default Create
