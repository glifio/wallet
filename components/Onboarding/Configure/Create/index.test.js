import { cleanup, render, screen, act, fireEvent } from '@testing-library/react'
import composeMockAppTree from '../../../../test-utils/composeMockAppTree'
import { mockRouterPush } from '../../../../test-utils/mocks/mock-routing'
import { flushPromises } from '../../../../test-utils'

import Create from '.'

describe('Create seed phrase configuration', () => {
  beforeAll(() => {
    global.URL.createObjectURL = jest.fn()
  })
  afterEach(() => {
    jest.clearAllMocks()
    cleanup()
  })
  test('it renders correctly', () => {
    const { Tree } = composeMockAppTree('postOnboard')

    const { container } = render(
      <Tree>
        <Create initialWalkthroughStep={3} />
      </Tree>
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  test('it sends the user to wallet view, with a wallet in state upon successful config', async () => {
    const mockWalletProviderDispatch = jest.fn()
    const { Tree, store } = composeMockAppTree('preOnboard', {
      walletProviderDispatch: mockWalletProviderDispatch
    })

    const { container } = render(
      <Tree>
        <Create initialWalkthroughStep={3} />
      </Tree>
    )

    await act(async () => {
      fireEvent.click(screen.getByText('Next'))
      await flushPromises()
    })
    expect(container.firstChild).toMatchSnapshot()
    expect(mockRouterPush).toHaveBeenCalledWith('/home?network=t')
    expect(mockWalletProviderDispatch.mock.calls[0][0].type).toBe(
      'CREATE_WALLET_PROVIDER'
    )
    expect(store.getState().wallets.length).toBe(1)
  })
})
