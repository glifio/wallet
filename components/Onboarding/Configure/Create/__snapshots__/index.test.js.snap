// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create seed phrase configuration it renders correctly 1`] = `
.c0 {
  min-width: 0;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  min-width: 0;
  box-sizing: border-box;
  border: 1px solid;
  border-radius: 4px;
  border-width: 1px;
  padding: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  max-width: 800px;
  min-height: 240px;
  overflow: hidden;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c4 {
  min-width: 0;
  box-sizing: border-box;
  border-width: 0.1875rem;
  border-style: solid;
  color: #000;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 48px;
  height: 48px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c6 {
  min-width: 0;
  box-sizing: border-box;
  margin-left: 24px;
  margin-top: 0;
  margin-bottom: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c8 {
  min-width: 0;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c9 {
  min-width: 0;
  box-sizing: border-box;
  border-radius: 100px;
  margin-left: 4px;
  margin-right: 4px;
  background-color: #1AD08F;
  display: inline-block;
  width: 8px;
  height: 8px;
}

.c10 {
  min-width: 0;
  box-sizing: border-box;
  margin-top: 16px;
  margin-bottom: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 64px;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-flex: 99;
  -webkit-flex-grow: 99;
  -ms-flex-positive: 99;
  flex-grow: 99;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c12 {
  min-width: 0;
  box-sizing: border-box;
  margin-top: 8px;
}

.c20 {
  min-width: 0;
  box-sizing: border-box;
  margin-top: 48px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c5 {
  font-family: "RT-Alias-Medium","system-ui","Segoe UI","Roboto",Helvetica;
  font-weight: 700;
  font-size: 1.5rem;
}

.c13 {
  cursor: pointer;
  border: 1px solid #262626;
  background-color: transparent;
  border-color: #262626;
  color: #262626;
  font-size: 1.125rem;
  -webkit-transition: 0.18s ease-in-out;
  transition: 0.18s ease-in-out;
  border-radius: 4px;
  margin-left: 8px;
  margin-right: 8px;
  padding: 16px;
  height: -webkit-max-content;
  height: -moz-max-content;
  height: max-content;
  border-width: 1px;
  border-radius: 4px;
}

.c13:hover {
  opacity: 0.8;
}

.c21 {
  cursor: pointer;
  border: 1px solid #262626;
  background-color: transparent;
  border-color: #262626;
  color: #262626;
  font-size: 1.125rem;
  -webkit-transition: 0.18s ease-in-out;
  transition: 0.18s ease-in-out;
  border-radius: 4px;
  margin-right: 8px;
  padding: 16px;
  border-width: 1px;
  border-radius: 4px;
}

.c21:hover {
  opacity: 0.8;
}

.c22 {
  cursor: pointer;
  border: 1px solid #1AD08F;
  background-color: #1AD08F;
  border-color: #1AD08F;
  color: #08442F;
  font-size: 1.125rem;
  -webkit-transition: 0.18s ease-in-out;
  transition: 0.18s ease-in-out;
  border-radius: 4px;
  margin-left: 8px;
  padding: 16px;
  border-width: 1px;
  border-radius: 4px;
}

.c22:hover {
  opacity: 0.8;
}

.c11 {
  font-size: 1.5rem;
  font-weight: 400;
  line-height: 1.2;
  font-family: RT-Alias-Grotesk;
  margin: 0;
}

.c7 {
  color: #262626;
  font-size: 1.25rem;
  font-weight: 400;
  line-height: 1.4;
  font-family: RT-Alias-Grotesk;
  margin-right: 8px;
  margin-top: 0;
  margin-bottom: 0;
}

.c18 {
  color: #1AD08F;
  font-size: 1.25rem;
  font-weight: 400;
  line-height: 1.4;
  font-family: RT-Alias-Grotesk;
}

.c2 {
  list-style: none;
  margin: 0;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  border-color: silver;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c16 {
  min-width: 0;
  box-sizing: border-box;
  border: 1px solid;
  border-color: #1AD08F;
  border-radius: 100px;
  padding-top: 8px;
  padding-bottom: 8px;
  margin-top: 8px;
  margin-bottom: 8px;
  margin-right: 8px;
  background-color: #1AD08F;
  width: 160px;
  max-width: 480px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c17 {
  min-width: 0;
  box-sizing: border-box;
  border-radius: 100px;
  margin-left: 8px;
  margin-right: 8px;
  background-color: #08442F;
  height: 32px;
  width: 48px;
  font-size: 1.25rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-flex: 2;
  -webkit-flex-grow: 2;
  -ms-flex-positive: 2;
  flex-grow: 2;
}

.c19 {
  min-width: 0;
  box-sizing: border-box;
  border-radius: 100px;
  border: 0;
  background-color: transparent;
  color: darkgreen;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 32px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: left;
  font-size: 1.25rem;
  line-height: 2;
}

.c15 {
  min-width: 0;
  box-sizing: border-box;
  margin-top: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  list-style: none;
}

.c14 {
  cursor: pointer;
  background-color: transparent;
  border-color: #262626;
  color: #262626;
  font-size: 1.125rem;
  -webkit-transition: 0.18s ease-in-out;
  transition: 0.18s ease-in-out;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 4px;
  padding: 16px;
  height: -webkit-max-content;
  height: -moz-max-content;
  height: max-content;
  border: 1px solid;
  border-radius: 4px;
}

.c14:hover {
  opacity: 0.8;
}

@media screen and (min-width:40em) {
  .c10 {
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
  }
}

@media screen and (min-width:40em) {
  .c12 {
    margin-top: 0;
  }
}

@media screen and (min-width:40em) {
  .c15 {
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
  }
}

<div
  class="c0"
  display="flex"
>
  <div
    class="c1"
    display="flex"
    overflow="hidden"
    width="100%"
  >
    <ul
      class="c2"
      display="flex"
      width="100%"
    >
      <li
        class="c3"
        display="flex"
      >
        <div
          class="c4"
          color="#000"
          display="flex"
          size="6"
        >
          <h3
            class="c5"
          >
            Sp
          </h3>
        </div>
      </li>
      <li
        class=""
      >
        <div
          class="c6"
          display="flex"
        >
          <p
            class="c7"
            color="core.nearblack"
            font-family="RT-Alias-Grotesk"
            font-size="3"
            font-weight="400"
          >
            Step 
            3
          </p>
          <div
            class="c8"
            display="flex"
          >
            <div
              class="c9"
              display="inline-block"
              height="2"
              width="2"
            />
          </div>
          <div
            class="c8"
            display="flex"
          >
            <div
              class="c9"
              display="inline-block"
              height="2"
              width="2"
            />
          </div>
          <div
            class="c8"
            display="flex"
          >
            <div
              class="c9"
              display="inline-block"
              height="2"
              width="2"
            />
          </div>
        </div>
      </li>
    </ul>
    <div
      class="c10"
      display="flex"
    >
      <h2
        class="c11"
        font-family="RT-Alias-Grotesk"
        font-size="4"
        font-weight="400"
      >
        Success! Please click 'Next' to access your wallet.
      </h2>
      <div
        class="c12"
      >
        <button
          class="c13"
          font-size="3"
          height="max-content"
          type="button"
        >
          Copy
        </button>
        <a
          class="c14"
          download="dontlookhere.txt"
          font-size="3"
          height="max-content"
          role="button"
        >
          Download
        </a>
      </div>
    </div>
    <div
      class="c15"
      display="flex"
    >
      <li
        class=""
      >
        <div
          class="c16"
          width="9"
        >
          <div
            class="c17"
            font-size="3"
            height="5"
            width="6"
          >
            <p
              class="c18"
              color="status.success.background"
              font-family="RT-Alias-Grotesk"
              font-size="3"
              font-weight="400"
            >
              1
            </p>
          </div>
          <div
            class="c19"
            color="darkgreen"
            display="flex"
            font-size="3"
            height="5"
            width="100%"
          >
            slender
          </div>
        </div>
      </li>
      <li
        class=""
      >
        <div
          class="c16"
          width="9"
        >
          <div
            class="c17"
            font-size="3"
            height="5"
            width="6"
          >
            <p
              class="c18"
              color="status.success.background"
              font-family="RT-Alias-Grotesk"
              font-size="3"
              font-weight="400"
            >
              2
            </p>
          </div>
          <div
            class="c19"
            color="darkgreen"
            display="flex"
            font-size="3"
            height="5"
            width="100%"
          >
            spread
          </div>
        </div>
      </li>
      <li
        class=""
      >
        <div
          class="c16"
          width="9"
        >
          <div
            class="c17"
            font-size="3"
            height="5"
            width="6"
          >
            <p
              class="c18"
              color="status.success.background"
              font-family="RT-Alias-Grotesk"
              font-size="3"
              font-weight="400"
            >
              3
            </p>
          </div>
          <div
            class="c19"
            color="darkgreen"
            display="flex"
            font-size="3"
            height="5"
            width="100%"
          >
            awkward
          </div>
        </div>
      </li>
      <li
        class=""
      >
        <div
          class="c16"
          width="9"
        >
          <div
            class="c17"
            font-size="3"
            height="5"
            width="6"
          >
            <p
              class="c18"
              color="status.success.background"
              font-family="RT-Alias-Grotesk"
              font-size="3"
              font-weight="400"
            >
              4
            </p>
          </div>
          <div
            class="c19"
            color="darkgreen"
            display="flex"
            font-size="3"
            height="5"
            width="100%"
          >
            chicken
          </div>
        </div>
      </li>
      <li
        class=""
      >
        <div
          class="c16"
          width="9"
        >
          <div
            class="c17"
            font-size="3"
            height="5"
            width="6"
          >
            <p
              class="c18"
              color="status.success.background"
              font-family="RT-Alias-Grotesk"
              font-size="3"
              font-weight="400"
            >
              5
            </p>
          </div>
          <div
            class="c19"
            color="darkgreen"
            display="flex"
            font-size="3"
            height="5"
            width="100%"
          >
            noise
          </div>
        </div>
      </li>
      <li
        class=""
      >
        <div
          class="c16"
          width="9"
        >
          <div
            class="c17"
            font-size="3"
            height="5"
            width="6"
          >
            <p
              class="c18"
              color="status.success.background"
              font-family="RT-Alias-Grotesk"
              font-size="3"
              font-weight="400"
            >
              6
            </p>
          </div>
          <div
            class="c19"
            color="darkgreen"
            display="flex"
            font-size="3"
            height="5"
            width="100%"
          >
            useful
          </div>
        </div>
      </li>
      <li
        class=""
      >
        <div
          class="c16"
          width="9"
        >
          <div
            class="c17"
            font-size="3"
            height="5"
            width="6"
          >
            <p
              class="c18"
              color="status.success.background"
              font-family="RT-Alias-Grotesk"
              font-size="3"
              font-weight="400"
            >
              7
            </p>
          </div>
          <div
            class="c19"
            color="darkgreen"
            display="flex"
            font-size="3"
            height="5"
            width="100%"
          >
            thank
          </div>
        </div>
      </li>
      <li
        class=""
      >
        <div
          class="c16"
          width="9"
        >
          <div
            class="c17"
            font-size="3"
            height="5"
            width="6"
          >
            <p
              class="c18"
              color="status.success.background"
              font-family="RT-Alias-Grotesk"
              font-size="3"
              font-weight="400"
            >
              8
            </p>
          </div>
          <div
            class="c19"
            color="darkgreen"
            display="flex"
            font-size="3"
            height="5"
            width="100%"
          >
            dentist
          </div>
        </div>
      </li>
      <li
        class=""
      >
        <div
          class="c16"
          width="9"
        >
          <div
            class="c17"
            font-size="3"
            height="5"
            width="6"
          >
            <p
              class="c18"
              color="status.success.background"
              font-family="RT-Alias-Grotesk"
              font-size="3"
              font-weight="400"
            >
              9
            </p>
          </div>
          <div
            class="c19"
            color="darkgreen"
            display="flex"
            font-size="3"
            height="5"
            width="100%"
          >
            tip
          </div>
        </div>
      </li>
      <li
        class=""
      >
        <div
          class="c16"
          width="9"
        >
          <div
            class="c17"
            font-size="3"
            height="5"
            width="6"
          >
            <p
              class="c18"
              color="status.success.background"
              font-family="RT-Alias-Grotesk"
              font-size="3"
              font-weight="400"
            >
              10
            </p>
          </div>
          <div
            class="c19"
            color="darkgreen"
            display="flex"
            font-size="3"
            height="5"
            width="100%"
          >
            bronze
          </div>
        </div>
      </li>
      <li
        class=""
      >
        <div
          class="c16"
          width="9"
        >
          <div
            class="c17"
            font-size="3"
            height="5"
            width="6"
          >
            <p
              class="c18"
              color="status.success.background"
              font-family="RT-Alias-Grotesk"
              font-size="3"
              font-weight="400"
            >
              11
            </p>
          </div>
          <div
            class="c19"
            color="darkgreen"
            display="flex"
            font-size="3"
            height="5"
            width="100%"
          >
            ritual
          </div>
        </div>
      </li>
      <li
        class=""
      >
        <div
          class="c16"
          width="9"
        >
          <div
            class="c17"
            font-size="3"
            height="5"
            width="6"
          >
            <p
              class="c18"
              color="status.success.background"
              font-family="RT-Alias-Grotesk"
              font-size="3"
              font-weight="400"
            >
              12
            </p>
          </div>
          <div
            class="c19"
            color="darkgreen"
            display="flex"
            font-size="3"
            height="5"
            width="100%"
          >
            explain
          </div>
        </div>
      </li>
      <li
        class=""
      >
        <div
          class="c16"
          width="9"
        >
          <div
            class="c17"
            font-size="3"
            height="5"
            width="6"
          >
            <p
              class="c18"
              color="status.success.background"
              font-family="RT-Alias-Grotesk"
              font-size="3"
              font-weight="400"
            >
              13
            </p>
          </div>
          <div
            class="c19"
            color="darkgreen"
            display="flex"
            font-size="3"
            height="5"
            width="100%"
          >
            version
          </div>
        </div>
      </li>
      <li
        class=""
      >
        <div
          class="c16"
          width="9"
        >
          <div
            class="c17"
            font-size="3"
            height="5"
            width="6"
          >
            <p
              class="c18"
              color="status.success.background"
              font-family="RT-Alias-Grotesk"
              font-size="3"
              font-weight="400"
            >
              14
            </p>
          </div>
          <div
            class="c19"
            color="darkgreen"
            display="flex"
            font-size="3"
            height="5"
            width="100%"
          >
            spot
          </div>
        </div>
      </li>
      <li
        class=""
      >
        <div
          class="c16"
          width="9"
        >
          <div
            class="c17"
            font-size="3"
            height="5"
            width="6"
          >
            <p
              class="c18"
              color="status.success.background"
              font-family="RT-Alias-Grotesk"
              font-size="3"
              font-weight="400"
            >
              15
            </p>
          </div>
          <div
            class="c19"
            color="darkgreen"
            display="flex"
            font-size="3"
            height="5"
            width="100%"
          >
            collect
          </div>
        </div>
      </li>
      <li
        class=""
      >
        <div
          class="c16"
          width="9"
        >
          <div
            class="c17"
            font-size="3"
            height="5"
            width="6"
          >
            <p
              class="c18"
              color="status.success.background"
              font-family="RT-Alias-Grotesk"
              font-size="3"
              font-weight="400"
            >
              16
            </p>
          </div>
          <div
            class="c19"
            color="darkgreen"
            display="flex"
            font-size="3"
            height="5"
            width="100%"
          >
            whisper
          </div>
        </div>
      </li>
      <li
        class=""
      >
        <div
          class="c16"
          width="9"
        >
          <div
            class="c17"
            font-size="3"
            height="5"
            width="6"
          >
            <p
              class="c18"
              color="status.success.background"
              font-family="RT-Alias-Grotesk"
              font-size="3"
              font-weight="400"
            >
              17
            </p>
          </div>
          <div
            class="c19"
            color="darkgreen"
            display="flex"
            font-size="3"
            height="5"
            width="100%"
          >
            glow
          </div>
        </div>
      </li>
      <li
        class=""
      >
        <div
          class="c16"
          width="9"
        >
          <div
            class="c17"
            font-size="3"
            height="5"
            width="6"
          >
            <p
              class="c18"
              color="status.success.background"
              font-family="RT-Alias-Grotesk"
              font-size="3"
              font-weight="400"
            >
              18
            </p>
          </div>
          <div
            class="c19"
            color="darkgreen"
            display="flex"
            font-size="3"
            height="5"
            width="100%"
          >
            peanut
          </div>
        </div>
      </li>
      <li
        class=""
      >
        <div
          class="c16"
          width="9"
        >
          <div
            class="c17"
            font-size="3"
            height="5"
            width="6"
          >
            <p
              class="c18"
              color="status.success.background"
              font-family="RT-Alias-Grotesk"
              font-size="3"
              font-weight="400"
            >
              19
            </p>
          </div>
          <div
            class="c19"
            color="darkgreen"
            display="flex"
            font-size="3"
            height="5"
            width="100%"
          >
            bus
          </div>
        </div>
      </li>
      <li
        class=""
      >
        <div
          class="c16"
          width="9"
        >
          <div
            class="c17"
            font-size="3"
            height="5"
            width="6"
          >
            <p
              class="c18"
              color="status.success.background"
              font-family="RT-Alias-Grotesk"
              font-size="3"
              font-weight="400"
            >
              20
            </p>
          </div>
          <div
            class="c19"
            color="darkgreen"
            display="flex"
            font-size="3"
            height="5"
            width="100%"
          >
            local
          </div>
        </div>
      </li>
      <li
        class=""
      >
        <div
          class="c16"
          width="9"
        >
          <div
            class="c17"
            font-size="3"
            height="5"
            width="6"
          >
            <p
              class="c18"
              color="status.success.background"
              font-family="RT-Alias-Grotesk"
              font-size="3"
              font-weight="400"
            >
              21
            </p>
          </div>
          <div
            class="c19"
            color="darkgreen"
            display="flex"
            font-size="3"
            height="5"
            width="100%"
          >
            country
          </div>
        </div>
      </li>
      <li
        class=""
      >
        <div
          class="c16"
          width="9"
        >
          <div
            class="c17"
            font-size="3"
            height="5"
            width="6"
          >
            <p
              class="c18"
              color="status.success.background"
              font-family="RT-Alias-Grotesk"
              font-size="3"
              font-weight="400"
            >
              22
            </p>
          </div>
          <div
            class="c19"
            color="darkgreen"
            display="flex"
            font-size="3"
            height="5"
            width="100%"
          >
            album
          </div>
        </div>
      </li>
      <li
        class=""
      >
        <div
          class="c16"
          width="9"
        >
          <div
            class="c17"
            font-size="3"
            height="5"
            width="6"
          >
            <p
              class="c18"
              color="status.success.background"
              font-family="RT-Alias-Grotesk"
              font-size="3"
              font-weight="400"
            >
              23
            </p>
          </div>
          <div
            class="c19"
            color="darkgreen"
            display="flex"
            font-size="3"
            height="5"
            width="100%"
          >
            punch
          </div>
        </div>
      </li>
      <li
        class=""
      >
        <div
          class="c16"
          width="9"
        >
          <div
            class="c17"
            font-size="3"
            height="5"
            width="6"
          >
            <p
              class="c18"
              color="status.success.background"
              font-family="RT-Alias-Grotesk"
              font-size="3"
              font-weight="400"
            >
              24
            </p>
          </div>
          <div
            class="c19"
            color="darkgreen"
            display="flex"
            font-size="3"
            height="5"
            width="100%"
          >
            frown
          </div>
        </div>
      </li>
    </div>
  </div>
  <div
    class="c20"
    display="flex"
    width="100%"
  >
    <button
      class="c21"
      font-size="3"
      type="button"
    >
      Back
    </button>
    <button
      class="c22"
      font-size="3"
      type="button"
    >
      Next
    </button>
  </div>
</div>
`;

exports[`Create seed phrase configuration it sends the user to wallet view, with a wallet in state upon successful config 1`] = `
.c0 {
  min-width: 0;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3 {
  font-size: 1.125rem;
  font-weight: 400;
  line-height: 1;
  font-family: RT-Alias-Grotesk;
  margin: 0;
  margin-top: 16px;
}

.c1 {
  display: inline-block;
  height: 48px;
  width: 48px;
  background-color: #5E26FF;
}

.c2 {
  display: inline-block;
  height: 24px;
  width: 24px;
  background-color: rgb(26,0,102);
  -webkit-animation-name: bvGDKI;
  animation-name: bvGDKI;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
}

@media screen and (min-width:40em) {

}

@media screen and (min-width:40em) {

}

@media screen and (min-width:40em) {

}

<div
  class="c0"
  display="flex"
>
  <span
    class="c1"
  >
    <span
      class="c2"
    />
  </span>
  <h4
    class="c3"
    font-family="RT-Alias-Grotesk"
    font-size="2"
    font-weight="400"
  >
    Loading...
  </h4>
</div>
`;
